### ISSUE TITLE
How does Response support returning to SSE

### ISSUE THREAD

**Issue by Dartly** [vania-dart/framework#89]
How does Response support returning to SSE

**Comment by Dartly** [comment]
I added the sse method
`import 'dart:convert';
import 'dart:io';
import 'dart:typed_data';
import 'package:meta/meta.dart';
import 'package:vania/src/http/response/stream_file.dart';

enum ResponseType {
  json,
  none,
  html,
  streamFile,
  download,
  sse,
}

class Response {
  @protected
  final ResponseType responseType;
  @protected
  final dynamic data;
  @protected
  final int httpStatusCode;
  @protected
  final Map<String, String> headers;

  Response({
    this.data,
    this.responseType = ResponseType.none,
    this.httpStatusCode = HttpStatus.ok,
    this.headers = const {},
  });

  void makeResponse(HttpResponse res) async {
    res.statusCode = httpStatusCode;
    if (headers.isNotEmpty) {
      headers.forEach((key, value) {
        res.headers.add(key, value);
      });
    }
    switch (responseType) {
      case ResponseType.json:
        res.headers.contentType = ContentType.json;
        try {
          res.write(jsonEncode(data));
        } catch (_) {
          res.write(jsonEncode(data.toString()));
        }
        await res.close();
        break;
      case ResponseType.html:
        res.headers.contentType = ContentType.html;
        res.write(data);
        await res.close();
        break;
      case ResponseType.streamFile:
        StreamFile? stream = StreamFile(
          fileName: data['fileName'],
          bytes: data['bytes'],
        ).call();
        if (stream == null) {
          res.headers.contentType = ContentType.json;
          res.write(jsonEncode({"message": "File not found"}));
          await res.close();
          break;
        }
        res.headers.contentType = stream.contentType;
        res.headers.contentLength = stream.length;
        await res.addStream(stream.stream!);
        await res.close();
        break;
      case ResponseType.download:
        StreamFile? downloadStream = StreamFile(
          fileName: data['fileName'],
          bytes: data['bytes'],
        ).call();
        if (downloadStream == null) {
          res.headers.contentType = ContentType.json;
          res.write(jsonEncode({"message": "File not found"}));
          await res.close();
          break;
        }
        res.headers.contentType = downloadStream.contentType;
        res.headers.contentLength = downloadStream.length;
        res.headers.add("Content-Disposition", downloadStream.contentDisposition);
        await res.addStream(downloadStream.stream!);
        await res.close();
        break;

      case ResponseType.sse:
        res.headers.contentType = ContentType.parse('text/event-stream');
        res.headers.add(HttpHeaders.cacheControlHeader, 'no-cache');
        res.headers.add(HttpHeaders.connectionHeader, 'keep-alive');
        res.headers.add(HttpHeaders.transferEncodingHeader, 'chunked');

        void writeSSE(String event, String data) {
          res.write('event: $event\n');
          res.write('data: $data\n\n');
        }

        await for (var event in data) {
          if (event is Map) {
            writeSSE(event['event'], jsonEncode(event['data']));
          } else {
            writeSSE('message', event.toString());
          }
          await res.flush();
        }

        await res.close();
        break;
      default:
        res.write(data);
        await res.close();
    }
  }

  static json(
      dynamic jsonData, [
        int statusCode = HttpStatus.ok,
      ]) =>
      Response(
        data: jsonData,
        responseType: ResponseType.json,
        httpStatusCode: statusCode,
      );

  static jsonWithHeader(
      dynamic jsonData, {
        int statusCode = HttpStatus.ok,
        Map<String, String> headers = const {},
      }) =>
      Response(
        data: jsonData,
        responseType: ResponseType.json,
        httpStatusCode: statusCode,
        headers: headers,
      );

  static html(
      dynamic htmlData, {
        Map<String, String> headers = const {},
      }) =>
      Response(
        data: htmlData,
        responseType: ResponseType.html,
        headers: headers,
      );

  static file(
      String fileName,
      Uint8List bytes, {
        Map<String, String> headers = const {},
      }) =>
      Response(
        data: {
          "fileName": fileName,
          "bytes": bytes,
        },
        responseType: ResponseType.streamFile,
        headers: headers,
      );

  static download(
      String fileName,
      Uint8List bytes, {
        Map<String, String> headers = const {},
      }) =>
      Response(
        data: {
          "fileName": fileName,
          "bytes": bytes,
        },
        responseType: ResponseType.download,
        headers: headers,
      );



  static sse(
      Stream<dynamic> eventStream, {
        int statusCode = HttpStatus.ok,
        Map<String, String> headers = const {},
      }) =>
      Response(
        data: eventStream,
        responseType: ResponseType.sse,
        httpStatusCode: statusCode,
        headers: headers,
      );
}`

**Comment by Dartly** [comment]
### I added the sse method
```dart
import 'dart:convert';
import 'dart:io';
import 'dart:typed_data';
import 'package:meta/meta.dart';
import 'package:vania/src/http/response/stream_file.dart';

enum ResponseType {
  ...
  sse,
}

class Response {
       ...
      case ResponseType.sse:
        res.headers.contentType = ContentType.parse('text/event-stream');
        res.headers.add(HttpHeaders.cacheControlHeader, 'no-cache');
        res.headers.add(HttpHeaders.connectionHeader, 'keep-alive');
        res.headers.add(HttpHeaders.transferEncodingHeader, 'chunked');

        void writeSSE(String data) {
         res.add(utf8.encode('data: $data\n\n'));
        }

        await for (var event in data) {
          writeSSE(jsonEncode(event));
          await res.flush();
        }

        await res.close();
        break;
      default:
        res.write(data);
        await res.close();
    }
  }

  ...
  static sse(
      Stream<dynamic> eventStream, {
        int statusCode = HttpStatus.ok,
        Map<String, String> headers = const {},
      }) =>
      Response(
        data: eventStream,
        responseType: ResponseType.sse,
        httpStatusCode: statusCode,
        headers: headers,
      );
}
```

**Comment by javad-zobeidi** [comment]
@Dartly  your code merged (#90)

 Thank you,
